### PR TITLE
WIP: update for safe verified GMI cuts paper

### PR DESCRIPTION
**Pull request by leoneifler**
This already looks quite nice, some remaining todos:

- [x]  update affiliations
- [x]  what to do with the copyright at the start of the scripts? leave and just update the year?
- [x] correct/fix link to exact SCIP version
- [ ] also reference other papers where VIPR is used?

The main new thing is `viprcomp.cpp` and the change in the build system since viprcomp uses soplex both for solving exact LPs as well as for many of its data structures. Therefore switching to CMake makes it a lot easier to compile. Installation instructions have been added. `viprchk.cpp` has been refactored (and some minor bugs were fixed). The `cert_spec` file is now also available in markdown (now that GitHub finally supports markdown in latex).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,38 +1,98 @@
...
-- [`vipr2html`](code/viprchk.cpp): A program that converts `.vipr` certificate files to a human readable HTML format (not recommended for large files).
+## Software
 
-The code should compile with most modern compilers supporting the C++11 standard.  Directory [`code`](code/) provides a basic [`makefile`](code/makefile).  We have successfully built the tools with GNU g++ version 5.4.0 under Ubuntu 16.04.4.  It depends on the [GNU Multiple Precision library](https://gmplib.org/).
+*VIPR* currently provides five C++ scripts, each being called from a terminal together with an appropriate `.vipr` certificate file:
 
+- `vprchck`: A program that verifies mixed-integer linear programming certificate files specified in the `.vipr` file format.
+- `vipr2html`: A program that converts `.vipr` certificate files to a human readable HTML format (not recommended for large files).
+- `viprttn`: A program that tightens and improves `.vipr` files, potentially reducing their size and allowing for easier checking.
+- `viprwidth`: A program that returns the cutwidth on derived constraints.
```
Comment (ambros-gleixner) [inline README.md]: - file missing?
- describe what is meant by cutwidth here

File: README.md [hunk 2]
```diff
@@ -1,38 +1,98 @@
...
+    0   Error (default)
+    1   Warning
+    2   Debug
+    3   Normal
+    4   High
+    5   Full
+
+For further information, consult the [SoPlex Documentation](https://soplex.zib.de/doc/html/).
+
+Debugmode is either set to `ON` or `OFF` (default).
```
Comment (ambros-gleixner) [inline README.md]: Add an example call at the end.

File: README.md [hunk 3]
```diff
@@ -1,38 +1,98 @@
...
 
 > Kevin K.H. Cheung, Ambros Gleixner, and Daniel E. Steffy: [Verifying Integer Programming Results](http://dx.doi.org/10.1007/978-3-319-59250-3_13). In: F. Eisenbrand and J. Koenemann, eds., Integer Programming and Combinatorial Optimization: 19th International Conference, IPCO 2017, pp. 148-160, 2017, [`doi:10.1007/978-3-319-59250-3_13`](http://dx.doi.org/10.1007/978-3-319-59250-3_13).
 
-This repository contains a detailed technical [specification of the certificate file format](http://rawgit.com/ambros-gleixner/VIPR/master/cert_spec_v1_0.html), [software](code/) to check, display, and compress certificate files, and [supplementary information](experiments/) on the computational experiments conducted for the article above.
+This repository contains a detailed technical specification of the certificate file format in [Version 1.0](cert_spec_v1_0.html) and [Version 1.1](cert_spec_v1_1.html), [software](code/) to check, display, compress, and complete certificate files, and [supplementary information](experiments/) on the computational experiments conducted for the article above.
```
Comment (ambros-gleixner) [inline README.md]: describe the extension of lin by incomplete and weak in v1.1

File: README.md [hunk 4]
```diff
@@ -1,38 +1,98 @@
...
 ## File format specification `.vipr`
 
-A conceptual description of the verified integer programming result (`.vipr`) file format is given in the article [above](http://nbn-resolving.de/urn:nbn:de:0297-zib-61044).  A more detailed technical specification is provided [here](http://rawgit.com/ambros-gleixner/VIPR/master/cert_spec_v1_0.html).
+A conceptual description of the verified integer programming result (`.vipr`) file format is given in the above articles.  A more detailed technical specification is provided [here](cert_spec_v1_1.html).
 
 A small example is given as [paper_eg3.vipr](code/paper_eg3.vipr).  Certificates for large MIP instances from the literature can be found as part of the [supplementary information](experiments/) of the article.
 
+## Installation
+
+The `vipr` scripts are compiled using [CMake](https://cmake.org/).
```
Comment (ambros-gleixner) [inline README.md:34]: remove old makefile

**Comment by ambros-gleixner** [comment]
- I revised the README a bit, can you check the diff? I removed the "---" in the README, because github otherwise displays two horizontal rules
- For copyright in existing files that have been modified add another line "Copyright (c) 2022 Zuse Institute Berlin", same in viprcomp and viprwidth
- some more todos in the comments

File: README.md (hunk 1 above)
Comment (leoneifler) [reply README.md]: ah, that file is actually something that Kevin wrote years ago but was never meant to be published I guess? I will remove it from the README

Comment (ambros-gleixner) [reply README.md]: Okay, when that is done I can also test the build process.

File: README.md (hunk 3 above)
Comment (leoneifler) [reply README.md]: I would also completely remove the .html files and change the links to markdown (which now supports mathjax on github as well, so it is more readable than the html I think

Comment (ambros-gleixner) [reply README.md]: Happily!

**Pull request merged**